caddy.go

```diff
@@ -253,12 +243,24 @@ type Server interface {
 	// connections.
 	Listen() (net.Listener, error)
 
+	// ListenPacket starts listening by creating a new packetconn
+	// and returning it. It does not start accepting
+	// connections.
+	ListenPacket() (net.PacketConn, error)
+
```

Comment:
It would probably be worth explaining in the godoc whether this is required for all server types to implement or if this can be a no-op for TCP servers, etc.